docs/source/notebook.rst

```diff
@@ -356,9 +358,14 @@ to remember are the following:
     is thus a shortcut for the sequence :kbd:`Shift-Enter`, :kbd:`Ctrl-m a`.
     (:kbd:`Ctrl-m a` adds a new cell above the current one.)
 
-* :kbd:`Esc` and :kbd:`Enter`: Command mode and edit mode
+* :kbd:`Esc` and :kbd:`Enter`: When the cursor is not displayed with the current
+    cell it is in *command mode*.
     In command mode, you can easily navigate around the notebook using keyboard
-    shortcuts. In edit mode, you can edit text in cells.
+    shortcuts.
+    When the cursor is displayed the cell is in *edit mode*, and you can modify it
+    content to edit text in cells.
+    In edit mode, :kbd:`Esc` forces the cell into command mode, and in command mode :kbd:`Enter`
+    forces it into edit mode (in edit mode, it simply inserts a newline).
```

Comment:
I agree with @takluyver's comments. It's probably  best to split Esc and Enter into two bullets.

Esc - enables Command Mode
Enter - enables Edit Mode